server/protocol.c

```diff
@@ -1466,22 +1120,196 @@ static void apply_server_config(request_rec *r)
     r->per_dir_config = r->server->lookup_defaults;
 }
 
+typedef enum {
+    http_error_none,
+    http_error_badprotocol,
+    http_error_reject09,
+    http_error_badmethod09,
+} http_error;
```

Comment:
The parts from the former `rrl_error` that apply to HTTP in general. Used when assigning a REQUEST bucket to a `request_rec`.